src/server/pfs/server/driver.go

```diff
@@ -4232,12 +4271,24 @@ func (d *driver) applyWrite(key string, records *pfs.PutFileRecords, tree hashtr
 					}
 				}
 
-				if err := tree.PutFileOverwrite(key, []*pfs.Object{{Hash: record.ObjectHash}}, record.OverwriteIndex, delta); err != nil {
-					return err
+				if record.ObjectHash != "" {
```

Comment:
A user overwriting a file that was copied from an output repo would not work right? The objects would be overwritten, but the blockrefs would remain unchanged. Get file would then get the content from the block refs. It seems like the only way to overwrite a copied file from an output repo would be through another copy.

Reply:
Actually this might work because you apply the overwrite to both objects and blockrefs regardless of the top level function called (which seems a little confusing to me). I'm not sure if that will work correctly with whatever functionality uses a non-zero overwrite index.

Reply:
`non-zero` overwrite is basically only used by `put file --overwrite` to try to skip uploading parts of the file. That only really applies when there are objects, because it checks the objects to see if it can deduplicate data. So I think it should be fine.